### ISSUE TITLE
Integrate Kubernetes Connector Support with Kaniko Image Builder

### ISSUE THREAD

**Issue by strickvl** [zenml-io/zenml#2217]
## Open Source Contributors Welcomed!
Please comment below if you would like to work on this issue!

### Contact Details [Optional]

support@zenml.io

### What happened?
ZenML currently lacks support for [Kubernetes connectors](https://docs.zenml.io/stacks-and-components/auth-management/kubernetes-service-connector) in the [Kaniko image builder stack component](https://docs.zenml.io/stacks-and-components/component-guide/image-builders/kaniko). This limitation hinders the ability to efficiently authenticate and interact with Kubernetes clusters for image building tasks using Kaniko.

### Task Description
Implement Kubernetes connector support within the Kaniko image builder stack component of ZenML. This enhancement will enable better integration and authentication with Kubernetes environments for image building processes, leveraging Kaniko's capabilities.

### Expected Outcome
- Successful integration of Kubernetes connector support with the Kaniko image builder in ZenML.
- Users should be able to seamlessly use Kaniko for image building tasks in Kubernetes environments.
- Ensure robust authentication and interaction capabilities with Kubernetes clusters through ZenML.

### Steps to Implement

- Develop and integrate Kubernetes connector functionalities within the Kaniko image builder component.
- Ensure the connector provides efficient and secure authentication with Kubernetes clusters.
- Test the integration extensively to ensure compatibility and functionality within various Kubernetes environments.
- Update the documentation to include instructions and use cases for using the Kubernetes connector with the Kaniko image builder.

### Additional Context
Adding Kubernetes connector support to the Kaniko image builder will significantly enhance the flexibility and usability of ZenML in cloud-native environments, especially for users who rely on Kubernetes for their MLOps workflows.

### Code of Conduct
- [ ] I agree to follow this project's Code of Conduct

**Comment by PierreYoon0129** [comment]
@strickvl Hi there! If you don't mind, would it be fine for me to deep-dive in this issue?

**Comment by strickvl** [comment]
Go for it! If you have any questions, let us know!

**Comment by adtech97** [comment]
@strickvl  I've landed here through a good first issue, if not worked on currently may I proceed with a deep dive on this issue ?

**Comment by htahir1** [comment]
@adtech97 yes go ahead!

**Comment by adtech97** [comment]
So before I start have a question about the target branch are we looking to merge this into a particular release branch or it is fine to Track the main branch as upstream?

**Comment by htahir1** [comment]
@adtech97 Branch off of `develop`.. the detailed documentation is here: https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md

**Comment by adtech97** [comment]
That's great thanks.

**Comment by adtech97** [comment]
@htahir1 Just posting the update for work done till now, I've completed the implementation from my side by this weekend, aiming for a PR. WIP for unit tests.

**Comment by bcdurak** [comment]
Do you have any updates on this @adtech97?

**Comment by adtech97** [comment]
Hey @bcdurak i have code changes in local will spin-up the tests and post the PR link by tommorow, apologies for delay :)

**Comment by bcdurak** [comment]
@adtech97, no problem at all. Let us know if we can help in any way.

**Comment by adtech97** [comment]
@bcdurak So for unit tests i was following existing cases from test/integration/* so these are normal functions, just curious if adding a test for this with pytest is okay?

**Comment by bcdurak** [comment]
@adtech97 Yes, we have unit tests in `tests/unit` and a bunch of integration tests in `tests/integration`. Feel free to follow the same pattern and add any tests that you see fit.

**Comment by adtech97** [comment]
For the issue, i have added my code changes as per my understanding of the issue and test to cover those in following PR https://github.com/zenml-io/zenml/pull/4070 please let me know how they look
